### PR TITLE
feat: Implement Elixir Binding

### PR DESCRIPTION
**Pull request by MRGRAVITY817**
This PR resolves issue #1444 

## The goal of this PR
- Implement Elixir binding for GlueSQL memory storage
- Build a good foundation to add more storages in near future

### CODE REVIEW

File: pkg/elixir/test/memory_storage_test.exs [hunk 1]
```diff
@@ -0,0 +1,167 @@
+defmodule GlueSQLTest do
+  use ExUnit.Case
+  doctest GlueSQL
+
+  alias GlueSQL.Storages.MemoryStorage
+  alias GlueSQL
```
Comment (dkuku) [inline pkg/elixir/test/memory_storage_test.exs:6]: This is not needed in elixir.

**Comment by dkuku** [comment]
I love this idea. Have you maybe checked serde_rustler for sending the data back to the beam?
That's the most updated fork I found
https://github.com/benhaney/serde_rustler/
I did try it in my first rust code and it looks promising - I can even receive elixir structs from rustler:
https://github.com/dkuku/fitparser_ex/

**Comment by MRGRAVITY817** [comment]
> I love this idea. Have you maybe checked serde_rustler for sending the data back to the beam instead of serializing it to json? That's the most updated fork I found https://github.com/benhaney/serde_rustler/ I did try it in my first rust code and it looks promising - I can even receive elixir structs from rustler: https://github.com/dkuku/fitparser_ex/

@dkuku Oh, actually the main reason that I didn't used `serde_rustler` was because original repo wasn't being maintained for a while. Thanks, I'll try update my code with this.

**Comment by lu-zero** [comment]
Why this cannot live in a separate repo?

Also, unreproducible binaries on the repo would get the crate flagged.

**Comment by MRGRAVITY817** [comment]
Hi @lu-zero !

> Why this cannot live in a separate repo?

Actually, it can live in a separate repo. 

I first thought it was a good idea to include this in this repo (since javascript/python version is also here as well), but that is not mandatory.

> Also, unreproducible binaries on the repo would get the crate flagged.

Yeah, this also makes me think if it should be created in separate repo. Thanks for the advice 👍

**Comment by MRGRAVITY817** [comment]
@panarch What are your thoughts about creating a separate repo for this rather than putting inside `pkg` folder?  

@lu-zero has pointed out some valid insight, and I also think that if we keep adding language bindings it would be too much of a burden for maintainer & reviewers because they also have to learn a new language.

**Comment by panarch** [comment]
@lu-zero raised a valid concern about increasing complexity in the gluesql repository with additional language bindings. Given the resource constraints, continuously managing new bindings like Elixir internally seems challenging.

I propose creating a separate repository for Elixir support to maintain a clean and focused main repository. Documentation can be linked from gluesql.org/docs for consistency.

@MRGRAVITY817, could you share your thoughts on this approach?

**Comment by MRGRAVITY817** [comment]
@panarch Yup, I'll just create a new repo and add a link in README once it's done. Thanks!

**Pull request closed without merging**